File: qdele-backend/src/main/java/br/com/rell/qdele_backend/entities/DatabaseConnection.java
package br.com.rell.qdele_backend.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.SQLRestriction;

import java.time.LocalDateTime;

@Entity
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "database_connection")
@SQLRestriction("deleted != false")
public class DatabaseConnection {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;
    @Column(nullable = false)
    private String url;
    @Column(nullable = false)
    private Integer port;
    @Column(nullable = false)
    private String username;
    @Column(nullable = false)
    private String password;

    @Enumerated(EnumType.STRING)
    @Column(name = "database_type", nullable = false)
    private DatabaseType databaseType;

    private String createdBy;
    private LocalDateTime createdDate;
    private String modifiedBy;
    private LocalDateTime modifiedDate;
    @Column(nullable = false, columnDefinition = "BOOLEAN DEFAULT FALSE")
    private Boolean deleted;
}

File: delfos-frontend/src/services/types/DatabaseConnection.ts
export interface DatabaseConnection {
    id?: number;
    name: string;
    url: string;
    port: number;
    username: string;
    password: string;
    databaseType: string;
    createdBy?: string | null;
    createdDate?: string | null;
    modifiedBy?: string | null;
    modifiedDate?: string | null;
    deleted?: boolean;
    databaseUrlCompletedUrl?: string;
}
  